lib/prometheus/que_stats.rb

```diff
@@ -17,19 +17,48 @@ def worker_stats
       end
     end
 
-    def job_stats(filter = nil)
-      filter = "WHERE #{filter}" if filter
+    # ApplicationJob did not have to be here, but it's just harder to test otherwise because of ApplicationJob#delete_duplicates
+    JOB_CLASSES = %w[ApplicationJob ProcessEntryJob ProcessIntegrationEntryJob UpdateJob].inspect.tr('"', "'").freeze
+
+    def job_stats(*filters)
+      filter = "WHERE #{filters.join(' AND ')}" if filters.presence
```

Comment:
Does that work?

```ruby
filters.each_with_object(scope) {|filter, condition| condition.where(filter) }
```

Reply:
I had to do something similar once but in BabySqueel and using OR instead of AND, but in case it helps:

https://github.com/3scale/porta/blob/ec3962a0b7de9357afa280bde39c90965a19e93b/app/models/cinstance.rb#L181-L188

Reply:
:+1: Yes and baby_squeel is installed

Reply:
IDK for sure why this was originally done in raw SQL. Maybe the intention was for it to depend as little as possible in the Rails environment and other dependencies of the application. Who knows?

However, if we want to change this, there's actually much more than we can do here. We can use Que's scopes: https://github.com/que-rb/que/blob/a2c9a2d7a11debee4ce5bd6121a4a5e0f6fd3323/lib/que/active_record/model.rb#L10-L23. Anyway, I'd prefer making this a "refactoring" PR following this one, so we can move on with configuring the dashboard and putting the new metrics in production ASAP.

Reply:
That's a good point 👍 It is probably safer to do it in 2 PRs to don't have too many risky changes in only 1.